src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -34,14 +45,24 @@
 }
 
 // Group
-.pf-l-toolbar__group:not(:last-child) {
+.pf-l-toolbar__group {
+  // adds top margin to all group divs
+  margin-top: var(--pf-l-toolbar__group--MarginTop);
   margin-right: var(--pf-l-toolbar__group--MarginRight);
+
+  .pf-l-toolbar__item {
+    margin-top: 0;
+    margin-right: 0;
+  }
 }
 
 // Item
 .pf-l-toolbar__item {
+  margin-top: var(--pf-l-toolbar__item--MarginTop);
+  margin-right: var(--pf-l-toolbar__item--MarginRight);
```

Comment:
This is creating a margin right after the last item when the item is a direct descendent of the toolbar element.

Reply:
- added `--pf-l-toolbar__group-item--MarginRight: var(--pf-global--spacer--sm);`
- added `.pf-l-toolbar__item:last-child {margin-right: var(--pf-l-toolbar__group-item--MarginRight-Zero);}` under parent selector `.pf-l-toolbar__group` in order zero out right margin from last item within a `toolbar__group`